src/util/measurement_utils.js

```diff
@@ -0,0 +1,327 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Utility functions to allow measurement of code paths without introducing
+// memory pressure via allocations. The measurement code is almost allocation
+// free! The only allocation made is a slot in map for every new tag introduced
+// (which mean only for the first measurement for a new tag). Tags are supposed
+// to identify a location in code or and operation which mean that all should be
+// allocated after enough time, after that the measurement process will be
+// allocation free.
+//
+// Calculating the metrics for the current measuerments and producing a report
+// is an allocaiton heavy operation in shouldbe done outside of paths we would
+//  like to measure.
+//
+// IMPORTANT: Any change introduced to this file should continue to uphole
```

Comment:
If that's such an important property of this module then you should use a different way to specify it - rename this file to `alloc_free_measurement_utils.js` and remove this comment (it's not the most inviting comment).

BTW for me the term instrumentation is more suitable than measurement. A measurement is a generic term that does not imply that you are measuring the code itself. See https://en.wikipedia.org/wiki/Instrumentation_(computer_programming).

Reply:
Ok, the naming thing makes sense I will change. Regarding the comment, I am not sure the name alone can convey the desired result, especially that the file contains also allocating function (massively allocating functions) to produce reports.

Reply:
I don't see how that comment helps more. I would still prefer to remove this comment because the tone of it is not sending the right message.

Reply:
And every instrumentation code will have a reporting function that is not part of the instrumentation code itself but only parses its info

Reply:
done